src/main.py

```diff
@@ -0,0 +1,130 @@
+from typing import Optional
+
+import torch.optim
+import pandas as pd
+import torch.nn as nn
+from torch.utils.data import DataLoader
+
+from model import PitchSeqNN
+from src.data.dataset import ComposerClassificationDataset
+
+BATCH_SIZE = 16
+N_EPOCHS = 100
+
+
+def prepare_dataset_pitch_only(dataset: ComposerClassificationDataset):
+    """
+    Prepares the dataset by selecting only the pitch information from the samples and converting it
+    into a format suitable for training.
+
+    Args:
+        dataset (ComposerClassificationDataset): An instance of YourDatasetClass containing samples with 'notes'
+                                    and 'composer' information.
+
+    Returns:
+        None: Modifies the 'samples' attribute of the dataset in-place.
+    """
+
+    def divide(lst):
+        return [num / 127.0 for num in lst]
+
+    df = pd.DataFrame(dataset.samples)
+    df = df[["notes", "composer"]]
+    # change classnames to numbers
+    df["composer"] = df["composer"].apply(lambda x: 0 if x == dataset.selected_composers[0] else 1)
+    # take only ['pitch'] column as an input tensor, normalize it to belong to (0, 1]
+    df["notes"] = df["notes"].apply(lambda x: x["pitch"])
+    df["notes"] = df["notes"].apply(divide)
+    print(df.groupby("composer").size())
+    samples = [(torch.tensor(row["notes"], dtype=torch.float32), torch.tensor(row["composer"])) for _, row in df.iterrows()]
+    while len(samples) % BATCH_SIZE != 0:
+        samples.pop(-1)
+    dataset.samples = samples
+
+
+def train():
+    """
+    Train the ComposerClassifier model using the ComposerClassificationDataset.
+
+    Returns:
+        PitchSeqNN: Trained model after completing the training loop.
+    """
+    # get train data
+    train_data = ComposerClassificationDataset(split="train")
+    prepare_dataset_pitch_only(train_data)
+    train_dataloader = DataLoader(train_data, batch_size=16, shuffle=True)
+    # initialize model
+    model = PitchSeqNN()
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.00001)
+    criterion = nn.CrossEntropyLoss()
+    # training loop
+    for epoch in range(N_EPOCHS):
+        running_loss = 0.0
+        for i, data in enumerate(train_dataloader, 0):
+            inputs, labels = data
+            pred = model(inputs)
+            loss = criterion(pred, labels)
+            loss.backward()
+            optimizer.step()
+
+            running_loss += loss.item()
+            # Periodically print the running loss for monitoring the training progress.
+            if i % 500 == 499:
+                print(f"[{epoch + 1}, {i + 1:5d}] loss: {running_loss / 500:.3f}")
+                running_loss = 0.0
```

Comment:
```suggestion
      running_loss = running_loss / len(train_dataloader)                                                                    
      val_loss = test_model(model)
      print(f"[{epoch + 1}] loss: {running_loss:.3f} val loss: {val_loss:.3f}")  
      running_loss = 0.0

```

This is a train loop monitoring method that I think would work better here. This suggestion is more of a pseudo-code, but here's the gist of what's the point:

1. Dataset and model are both small, so we can check metrics every epoch. (I'm not sure if it's clear that indent in my suggestion is supposed to be on the epoch level)
2. If possible, it's best to monitor the same metric on both splits, so the print every epoch could look like this `[epoch 76] training loss: 0.021, validation loss: 0.7`. You could also include accuracy here.

Reply:
Thank you!!